### DISCUSSION TITLE
Black outline when using WindowChromeBehavior

### DISCUSSION THREAD

**Discussion by MattiasOnline** [ControlzEx/ControlzEx#192] [Q&A]
Hello,

trying to use the WindowChromeBehavior but I am getting a black outline (1 px) left, right and bottom. It seems to be part of default rendering since it's color changes when I activate/deactivate the window. I'm on .Net Framework 4.8.1. What am I doing wrong?

```
<Window
    x:Class="WindowChromeBehaviorTest.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:i="http://schemas.microsoft.com/xaml/behaviors"
    xmlns:ctzex="urn:controlzex"
    Title="MainWindow" Height="450" Width="800">
    <i:Interaction.Behaviors>
        <ctzex:WindowChromeBehavior
            ResizeBorderThickness="30" 
            EnableMinimize="True"
            EnableMaxRestore="True"
            KeepBorderOnMaximize="True" />
    </i:Interaction.Behaviors>

    <Border BorderThickness="3" BorderBrush="Pink">
        <Label
            Background="LightBlue"
            HorizontalAlignment="Center"
            VerticalAlignment="Center"
            Content="CLIENT AREA" />
    </Border>
</Window>

```

**Comment by batzen** [comment]
It's difficult to tell without a screenshot.
On Windows 11 i just get the regular window border around the window.

**Reply by MattiasOnline** [reply]
Screenshot here. I updated the xaml to add a border to make it clearer:

![image](https://github.com/ControlzEx/ControlzEx/assets/71459346/f4cafc07-1392-4a2e-9a47-9b0c22419c50)

**Reply by batzen** [reply]
That's strange.
Which windows version are you using?

**Reply by MattiasOnline** [reply]
Ah yes, I'm on Windows 10. I just tested it on Windows 11 and the artifact is not there. It would be nice if it worked on Windows 10 as well.

**Reply by batzen** [reply]
Which exact version of Windows 10 are you using?

**Reply by MattiasOnline** [reply]
Microsoft Windows 10 Pro, 10.0.19045 version 19045.

**Reply by batzen** [reply]
Will try to repro tomorrow using a Windows 10 VM.

**Reply by batzen** [reply]
That's just the regular Windows window border appearing.
On my test machine it look like this:
![image](https://github.com/ControlzEx/ControlzEx/assets/1971030/f81a206e-498c-48bc-a335-5585fee17f8d)
The color of the border is different because the Windows accent color is different from yours.

If you don't want that window border you can set `WindowStyle=None`, but you will loose the minimize/restore animation by doing so.
If you want the border to go all around your window you can also use the `GlowWindowBehavior`.
Or you can inherit your window from `WindowChromeWindow` and copy the style/template from https://github.com/ControlzEx/ControlzEx/blob/bf293f2732bea3ffa14b285d9270e597a31d24b9/src/ControlzEx.Showcase/App.xaml#L24 as a starting point.
The reason why the default window border only is at three sides and not on top is because you usually have the titlebar there.
In Windows 11 they fixed that mess and stopped relying on the titlebar to draw something that looks like a border, but isn't/wasn't.

**Reply by MattiasOnline** [reply]
I don't get the artifact when just using Microsoft's WindowChrome, so a bit surprised I'm getting it with ControlzEx. Anyway, thank you for looking into it, at least I'm not missing something.

**Reply by batzen** [reply]
I now know why this happens with ControlzEx but not with WindowChrome.
ControlzEx does not extend the glass frame whereas WindowChrome does.
As i am currently adding support for Backdrop (Mica etc.) to ControlzEx and doing so requires extending the glass frame you will be able to get rid of those borders either in version 7 or version 8 of ControlzEx.

**Reply by MattiasOnline** [reply]
That's great to hear! I will keep an eye out and test the upcoming versions.
